### PR TITLE
Speed up daemon startup by 15%

### PR DESCRIPTION
**Pull request by hughsie**
During my fwupd startup fu_plugin_has_custom_flag gets called 21 times
which causes all HWIDs to be enumerated with 346 calls to the quite
expensive fu_context_lookup_quirk_by_id() function.

Move the flag to a private hashset and enumerate the HWIDs only during
startup. There's nothing plugin specific about them anyway...

Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [ ] Code fix
- [ ] Feature
- [ ] Documentation

### CODE REVIEW

**Review by superm1 — APPROVED** [review]

File: libfwupdplugin/fu-plugin.c [hunk 1]
```diff
@@ -611,30 +611,12 @@ gboolean
 fu_plugin_has_custom_flag(FuPlugin *self, const gchar *flag)
 {
 	FuPluginPrivate *priv = GET_PRIVATE(self);
-	GPtrArray *guids;
 
 	g_return_val_if_fail(FU_IS_PLUGIN(self), FALSE);
 	g_return_val_if_fail(flag != NULL, FALSE);
 
 	/* never set up, e.g. in tests */
```
Comment (YehezkelShB) [inline libfwupdplugin/fu-plugin.c:618]: I know I'm late to this party, but the comments seems irrelevant for the code now (and raises the question if tests should've been handled differently in the new change too)
